hack/util.sh

```diff
@@ -74,6 +74,7 @@ function configure_os_server {
 	openshift start \
 	--write-config=${SERVER_CONFIG_DIR} \
 	--create-certs=false \
+	--dns="tcp://${API_HOST}:53" \
```

Comment:
The reason you can change the master address is because our config doesn't
cleanly handle the fact that you don't want 3 master configs, one for each
master.  You want one master config that only differs between the three.

On Thu, Jun 2, 2016 at 10:16 AM, David Eads <notifications@github.com>
wrote:

> In hack/util.sh
> <https://github.com/openshift/origin/pull/9117#discussion_r65547256>:
>
> > @@ -74,6 +74,7 @@ function configure_os_server {
> >  	openshift start \
> >  	--write-config=${SERVER_CONFIG_DIR} \
> >  	--create-certs=false \
> > +	--dns="tcp://${API_HOST}:53" \
>
> Those special cases won't die, because you still need them.
>
> They don't make any sense as written. You can override the DNS port for an
> API server, but only if you start openshift start master api, not if you
> do openshift start master, but the kube master address, that can change
> for both openshift start master api and openshift start master, but if
> you're running an openshift start master controllers you have to specify
> the listenaddress, but that setting will have no affect if you're running openshift
> start master?
>
> These things are virtually impossible to reason about ahead of time. I
> understand why they exist (port and/or interface deconflicting), but in
> their current state its a bit insane.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9117/files/c2cd1cdedd832be748e0fdcc26f7d63bd0d75d72#r65547256>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p5hhufrnPybkTWppNkaQAJSqIz4xks5qHuW_gaJpZM4Ir2aW>
> .
>